### ISSUE TITLE
Guide: Enums

### ISSUE THREAD

**Issue by nburns** [rmosolgo/graphql-ruby#3625]
We have an enum that's driven from data in the database, is there anyway to make graphql lookup the data each query?

What we're doing right now is something like:
```ruby
module Types
  class UserType < Base::BaseEnum
    description 'all user types'
    UserType.where(visible: true).each { |t| value(t.id, value: t.id) }
  end
end
```
(and dynamically changing the visible attribute of a particular user type so it would be included/excluded in the enum)

It looks like it requires a process restart or some hackery with the base class to update these values?

I'd be willing to submit a PR if changing this functionality is something that's desired, but I'm thinking that representing this data as an enum was not what enums were intended for?

**Comment by rmosolgo** [comment]
👋  thanks for the great question. Here's a previous answer to this question, which still holds true: https://github.com/rmosolgo/graphql-ruby/issues/2172#issuecomment-805872160

TL;DR: There's not a good way to modify Enums (particularly removing values) after the application boots, but I'm still planning on looking into it as part of #3622. If you're interested in opening a PR, I'd be happy to help out along the way! I have a suggestion for implementation the comment linked above, but anything that _works_ works for me!

**Comment by nburns** [comment]
Great I'll take a look! Thanks for your time and detailed response (should've checked the history) 😄
